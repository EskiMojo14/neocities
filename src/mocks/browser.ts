import { http, HttpResponse } from "msw";
import { setupWorker } from "msw/browser";
import pages from "./graph.json";

export const worker = setupWorker(
  http.get(/localhost:\d+\/___graph.json/, () => HttpResponse.json(pages)),
);
