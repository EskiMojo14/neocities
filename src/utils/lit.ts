import type { Properties } from "csstype";
import type { CSSResultGroup } from "lit";
import { css } from "lit";
import type { StyleInfo } from "lit-html/directives/style-map.js";
import { styleMap as _styleMap } from "lit-html/directives/style-map.js";

declare module "csstype" {
  // eslint-disable-next-line @typescript-eslint/consistent-indexed-object-style
  interface Properties {
    [variable: `--${string}`]: StyleInfo[string];
  }
}

export function styleMap(props: Properties) {
  return _styleMap(props as never);
}

export function rtl(styles: CSSResultGroup): CSSResultGroup {
  return css`
    :dir(rtl) &,
    &:dir(rtl) {
      ${styles};
    }
  `;
}

export function ensureDir(dir: string): "ltr" | "rtl" | "auto" {
  return dir === "rtl" || dir === "ltr" ? dir : "auto";
}
