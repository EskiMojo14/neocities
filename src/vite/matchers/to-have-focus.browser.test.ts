import { html } from "lit";
import { expect } from "vitest";
import { page } from "vitest/browser";
import { it } from "../utils.browser.ts";

it("should pass without shadow root", async () => {
  page.render(html`<button>Button</button>`);
  const button = page.getByRole("button");
  await button.click();
  expect(button.element()).toHaveFocus();
});

it("should pass with shadow root", async () => {
  const lightContainer = document.body.appendChild(
    document.createElement("div"),
  );
  const shadow = lightContainer.attachShadow({ mode: "open" });
  const shadowContainer = shadow.appendChild(document.createElement("div"));
  page.render(html`<button>Button</button>`, { container: shadowContainer });

  const button = page.getByRole("button");
  await button.click();
  expect(button.element()).toHaveFocus();
});

it("should fail without shadow root", async () => {
  page.render(html`<button>Button</button><input />`);
  const button = page.getByRole("button");
  await button.click();
  expect(() =>
    expect(page.getByRole("textbox").element()).toHaveFocus(),
  ).toThrowErrorMatchingSnapshot();
  expect(() =>
    expect(button.element()).not.toHaveFocus(),
  ).toThrowErrorMatchingSnapshot();
});

it("should fail with shadow root", async () => {
  const lightContainer = document.body.appendChild(
    document.createElement("div"),
  );
  const shadow = lightContainer.attachShadow({ mode: "open" });
  const shadowContainer = shadow.appendChild(document.createElement("div"));
  page.render(html`<button>Button</button><input />`, {
    container: shadowContainer,
  });

  const button = page.getByRole("button");
  await button.click();
  expect(() =>
    expect(page.getByRole("textbox").element()).toHaveFocus(),
  ).toThrowErrorMatchingSnapshot();
  expect(() =>
    expect(button.element()).not.toHaveFocus(),
  ).toThrowErrorMatchingSnapshot();
});

it("should fail with invalid element", () => {
  expect(() => {
    expect(null).toHaveFocus();
  }).toThrowErrorMatchingSnapshot();
});
