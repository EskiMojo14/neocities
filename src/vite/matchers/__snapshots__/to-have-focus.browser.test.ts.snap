// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should fail with shadow root 1`] = `
[Error: [2mexpect([22m[31melement[39m[2m).toHaveFocus()[22m
Expected element with focus:
  [32m<input />[39m
Received element with focus:
  [31m<button>
  Button
</button>[39m]
`;

exports[`should fail with shadow root 2`] = `
[Error: [2mexpect([22m[31melement[39m[2m).not.toHaveFocus()[22m
Received element is focused:
  [31m<button>
  Button
</button>[39m]
`;

exports[`should fail without shadow root 1`] = `
[Error: [2mexpect([22m[31melement[39m[2m).toHaveFocus()[22m
Expected element with focus:
  [32m<input />[39m
Received element with focus:
  [31m<button>
  Button
</button>[39m]
`;

exports[`should fail without shadow root 2`] = `
[Error: [2mexpect([22m[31melement[39m[2m).not.toHaveFocus()[22m
Received element is focused:
  [31m<button>
  Button
</button>[39m]
`;
